### ISSUE TITLE
Some[A] has no available apply methods

### ISSUE THREAD

**Issue by mxl** [kxbmap/configs#16]
Compilation fails with error:

> Some[A] has no available apply methods

for configuration case classes:
```
case class TestConfiguration(test2: Option[Test2Configuration])

case class Test2Configuration(testOption: Option[Int])
```

Compilation succeeds after
- renaming `testOption` to `testoption`
- or changing `testOption` type to `Int` 
- or moving `testOption` to `TestConfiguration`.

So this issue happens when you have camel case `Option` property in second or deeper level case class.

**Comment by AlixBa** [comment]
same here

```scala
// OK
case class Config(parent: Config.Parent)
object Config {
  import configs.syntax._
  case class Parent(child: Parent.Child)
  object Parent {
    case class Child()
  }
  lazy val configObject: Config = get[Config]("").value
}

// add A to parent param name - OK
case class Config(parentA: Config.Parent)
object Config {
  import configs.syntax._
  case class Parent(child: Parent.Child)
  object Parent {
    case class Child()
  }
  lazy val configObject: Config = get[Config]("").value
}

// add A to child param name (without the one on parent) - OK
case class Config(parent: Config.Parent)
object Config {
  import configs.syntax._
  case class Parent(childA: Parent.Child)
  object Parent {
    case class Child()
  }
  lazy val configObject: Config = get[Config]("").value
}

// add A to both parent & child param name - FAILS
case class Config(parentA: Config.Parent)
object Config {
  import configs.syntax._
  case class Parent(childA: Parent.Child)
  object Parent {
    case class Child()
  }
  lazy val configObject: Config = get[Config]("").value
}
```

**Comment by kxbmap** [comment]
Version 0.4.3 that is fixed this issue was published for Scala 2.11 and 2.12.0-RC1.
They will sync to Maven Central in few hours.
